gpcontrib/gp_orca/gporca.c

```diff
@@ -19,10 +91,34 @@ _PG_init(void)
 
 	if (!IS_QUERY_DISPATCHER())
 		return;
+
+	if (Gp_role == GP_ROLE_DISPATCH)
+	{
+		/* Initialize GPOPT */
+		OptimizerMemoryContext = AllocSetContextCreate(TopMemoryContext,
+													"GPORCA Top-level Memory Context",
+													ALLOCSET_DEFAULT_MINSIZE,
+													ALLOCSET_DEFAULT_INITSIZE,
+													ALLOCSET_DEFAULT_MAXSIZE);
+
+		InitGPOPT();
```

Comment:
Now the `InitGPOPT` function is called before the `GPMemoryProtect_Init` function is called?